src/resources/views/crud/inc/form_fields_script.blade.php

```diff
@@ -0,0 +1,159 @@
+<script>
+    /**
+     * A front-end representation of a Backpack field, with its main components.
+     *
+     * Makes it dead-simple for the developer to perform the most common
+     * javascript manipulations, and makes it easy to do custom stuff
+     * too, by exposing the main components (name, wrapper, input).
+     */
+    class CrudField {
+        constructor(name) {
+            this.name = name;
+            this.wrapper = $(`[bp-field-name*="${name}"][bp-field-wrapper]`).first();
+
+            if (this.wrapper.length === 0) {
+                console.error(`CrudField error! Could not select WRAPPER for "${this.name}"`);
+            }
+
+            this.type = this.wrapper.attr('bp-field-type');
+            this.$input = this.mainInput;
+            this.input = this.$input[0];
+
+            // Validate that the field has been found
+            if(!this.input) {
+                console.error(`CrudField error! Could not select INPUT for "${this.name}"`);
+            }
+        }
+
+        get value() {
+            let value = this.input.value;
+
+            // Parse the value if it's a number
+            if (value.length && !isNaN(value)) {
+                value = Number(value);
```

Comment:
@promatik are you sure this doesn't have unintended consequences? Here's what @pxpm had to say about it on Discord:

> I think I found a quite big inconsistency here:
> - In a text field, if you want to validate the length of the text you would do something like: `field.value.length <= 10`, it works as long as you write atleast one letter, if you happen to use all numbers like: 12345 it will be converted to a real number, and `field.value.length` will return undefined ...

Sounds to me like we have a reason NOT to convert to Number?

Reply:
I think there is also a good reason to convert the "real numbers" into numbers, so I proposed that we add a `parseIntegers` or something as a method in `CrudField` so if I am targetting a `number` field I am sure I will use `crud.field('number').change(function(){}).parseIntegers()`